productcat/src/main/java/com/visenze/productcat/android/data/DataCollection.java

```diff
@@ -0,0 +1,70 @@
+package com.visenze.productcat.android.data;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.telephony.TelephonyManager;
+
+import androidx.core.app.ActivityCompat;
+
+import com.visenze.productcat.android.model.DeviceInfo;
+import com.visenze.productcat.android.util.SecurityHelper;
+
+import java.io.UnsupportedEncodingException;
+
+public class DataCollection implements GetGAIDTask.OnTaskSuccess {
+
+    private GetGAIDTask gaidTask;
+
+    private String gaid;
+    private String didmd5;
+
+    public DataCollection(Context context) {
+
+        generateDoubleHashIMEI(context);
+
+        gaidTask = new GetGAIDTask(context, this);
+        gaidTask.execute();
+
+
+    }
+
+    private void generateDoubleHashIMEI(Context context) {
```

Comment:
rename this to generateHashIMEI

Reply:
thanks. changed already. forgot to change previously ..